internal/venice-client-common/src/main/java/com/linkedin/venice/schema/rmd/RmdUtils.java

```diff
@@ -0,0 +1,101 @@
+  public static ByteBuffer serializeRmdRecord(final Schema valueSchema, GenericRecord rmdRecord) {
+    byte[] rmdBytes = getRmdSerializer(valueSchema).serialize(rmdRecord, AvroSerializer.REUSE.get());
+    return ByteBuffer.wrap(rmdBytes);
+  }
+
+  private static RecordSerializer<GenericRecord> getRmdSerializer(final Schema valueSchema) {
+    return MapOrderingPreservingSerDeFactory.getSerializer(valueSchema);
+  }
+
+  public static GenericRecord deserializeRmdBytes(
+      final Schema writerSchema,
+      final Schema readerSchema,
+      final ByteBuffer rmdBytes) {
+    return getRmdDeserializer(writerSchema, readerSchema).deserialize(rmdBytes);
+  }
+
+  private static RecordDeserializer<GenericRecord> getRmdDeserializer(
+      final Schema writerSchema,
+      final Schema readerSchema) {
+    return MapOrderingPreservingSerDeFactory.getDeserializer(writerSchema, readerSchema);
+  }
+
+  /**
+   * Returns the type of union record given tsObject is. Right now it will be either root level long or
+   * generic record of per field timestamp.
+   * @param tsObject
+   * @return
+   */
+  public static RmdTimestampType getRmdTimestampType(Object tsObject) {
+    if (tsObject instanceof Long) {
+      return RmdTimestampType.VALUE_LEVEL_TIMESTAMP;
+    } else if (tsObject instanceof GenericRecord) {
+      return RmdTimestampType.PER_FIELD_TIMESTAMP;
+    } else {
+      throw new IllegalStateException("Unexpected type of timestamp object. Got timestamp object: " + tsObject);
+    }
+  }
+
+  public static long extractOffsetVectorSumFromRmd(GenericRecord replicationMetadataRecord) {
+    Object offsetVectorObject = replicationMetadataRecord.get(REPLICATION_CHECKPOINT_VECTOR_FIELD);
+    return sumOffsetVector(offsetVectorObject);
+  }
+
+  public static List<Long> extractTimestampFromRmd(GenericRecord replicationMetadataRecord) {
+    // TODO: This function needs a heuristic to work on field level timestamps. At time of writing, this function
+    // is only for recording the previous value of a record's timestamp, so we could consider specifying the incoming
+    // operation to identify if we care about the record level timestamp, or, certain fields and then returning an
+    // ordered
+    // list of those timestamps to compare post resolution. I hesitate to commit to an implementation here prior to
+    // putting
+    // the full write compute resolution into uncommented fleshed out glory. So we'll effectively ignore operations
+    // that aren't root level until then.
+    if (replicationMetadataRecord == null) {
+      return Collections.singletonList(0L);
+    }
+    Object timestampObject = replicationMetadataRecord.get(TIMESTAMP_FIELD_NAME);
+    RmdTimestampType rmdTimestampType = getRmdTimestampType(timestampObject);
+
+    if (rmdTimestampType == RmdTimestampType.VALUE_LEVEL_TIMESTAMP) {
+      return Collections.singletonList((long) timestampObject);
+    } else {
+      // not supported yet so ignore it
```

Comment:
I believe this is outdated. We need to scan the per field ts and extract the time stamp.